opensquirrel/common.py

```diff
@@ -49,3 +49,14 @@ def are_matrices_equivalent_up_to_global_phase(
     phase_difference = matrix_a[first_non_zero] / matrix_b[first_non_zero]
 
     return np.allclose(matrix_a, phase_difference * matrix_b)
+
+
+def is_identity_matrix(matrix: NDArray[np.complex128]) -> bool:
+    """Checks whether matrix is an identity matrix up to a global phase.
+
+    Args:
+        matrix: matrix to check.
+    Returns:
+        Whether matrix is an identity matrix up to a global phase.
+    """
+    return are_matrices_equivalent_up_to_global_phase(matrix, np.eye(matrix.shape[0]))
```

Comment:
Beautiful. Can we move this function to common.py?

Reply:
It is in common.py already